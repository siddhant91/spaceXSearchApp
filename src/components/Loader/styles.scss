@import '../../styles/main.scss';

.spacex-loader {
	&--wrapper {
		position: fixed;
		top: 0;
		right: 0;
		bottom: 0;
		left: 0;
		text-align: center;
		z-index: 100;
		overflow: hidden;
		background: rgba($spacex-color-white, 0.5);
	}
	&--container {
		background-color: transparent;
		border: $spacex-spacing-2 solid $spacex-color-secondary-green;
		border-top: $spacex-spacing-2 solid $spacex-color-primary-green;
		border-radius: 50%;
		width: 70px;
		height: 70px;
		vertical-align: middle;
		animation: spin 2s infinite linear;
		position: fixed;
		top: 50%;
		left: 45%;
		transform: translate(-50%, -50%);
	}
}

@keyframes spin {
	0% {
		transform: rotate(0deg);
	}
	100% {
		transform: rotate(360deg);
	}
}

@keyframes ellipsis1 {
	0% {
		transform: scale(0);
	}
	100% {
		transform: scale(1);
	}
}

@keyframes ellipsis2 {
	0% {
		transform: translate(0, 0);
	}
	100% {
		transform: translate(24px, 0);
	}
}

@keyframes ellipsis3 {
	0% {
		transform: scale(1);
	}
	100% {
		transform: scale(0);
	}
}
