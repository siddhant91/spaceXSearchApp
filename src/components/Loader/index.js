/**
 * Generic Loader Component
 */

// styles
import './styles.scss';

const Loader = () => {
	return (
		<div className="spacex-loader spacex-loader--wrapper">
			<div className="spacex-loader--container" />
		</div>
	);
};

Loader.propTypes = {};

Loader.defaultProps = {};

export default Loader;
