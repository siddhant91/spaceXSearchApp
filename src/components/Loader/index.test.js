import { render } from '@testing-library/react';
import Loader from './index';

it('Matches snapshot', () => {
	const { asFragment } = render(<Loader />);
	expect(asFragment()).toMatchSnapshot();
});
