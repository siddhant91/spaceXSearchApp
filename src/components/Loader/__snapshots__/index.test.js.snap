// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="spacex-loader spacex-loader--wrapper"
  >
    <div
      class="spacex-loader--container"
    />
  </div>
</DocumentFragment>
`;
